src/components/ImpactDialog.js

```diff
@@ -0,0 +1,194 @@
+        <>
+          <MuiDialogTitle disableTypography className={classes.root}>
+            <Typography variant="h6">
+              You just helped a shelter cat! Congrats!
+            </Typography>
+          </MuiDialogTitle>
+          <MuiDialogContent>
+            <div className={classes.centerImage}>
+              <RandomGif />
+            </div>
+            <Typography className={classes.typographySpacing}>
+              The more people who open tabs, the more good we can do! Get a
+              friend on board and we'll give a treat to another 10 shelter cats:
+            </Typography>
+            <div className={classes.shareContainer}>
+              <InviteFriends user={user} />
+              <SocialShare url="https://tab.gladly.io" />
+            </div>
+          </MuiDialogContent>
+          <MuiDialogActions>
+            <Button autoFocus onClick={buttonOnClick} color="primary">
+              DONE
+            </Button>
+          </MuiDialogActions>
+        </>
+      )
+      break
+    case 'claimReferralReward':
+      dialogContent = (
+        <>
+          <MuiDialogTitle disableTypography className={classes.root}>
+            <Typography variant="h6">
+              {`You just put ${referralImpact} cats on track for adoption!`}
+            </Typography>
+          </MuiDialogTitle>
+          <MuiDialogContent>
+            <div className={classes.centerImage}>
+              <RandomGif />
+            </div>
+            <Typography className={classes.typographySpacing}>
+              Congratulations! You're making a huge impact for these animals in
+              need. Want to help even more cats? Invite a few more friends!
+            </Typography>
+            <div className={classes.shareContainer}>
+              <InviteFriends user={user} />
+              <SocialShare url="https://tab.gladly.io" />
+            </div>
+          </MuiDialogContent>
+          <MuiDialogActions>
+            <Button autoFocus onClick={buttonOnClick} color="primary">
+              DONE
+            </Button>
+          </MuiDialogActions>
+        </>
+      )
+      break
+    default:
+      dialogContent = <div />
+      break
+  }
+
+  return (
+    <Dialog
+      onClose={onClose || buttonOnClick}
+      aria-labelledby="customized-dialog-title"
+      open={open}
+      className={classes.rootModal}
+    >
+      {dialogContent}
+    </Dialog>
+  )
+}
+
+ImpactDialog.displayName = 'ImpactDialog'
+ImpactDialog.propTypes = {
+  modalType: PropTypes.oneOf([
+    'confirmImpact',
+    'impactWalkMe',
+    'claimImpactReward',
+    'claimReferralReward',
+  ]).isRequired,
+  buttonOnClick: PropTypes.func,
+  onClose: PropTypes.func,
+  open: PropTypes.bool.isRequired,
+  // eslint-disable-next-line react/require-default-props
+  referralImpact: PropTypes.number,
+  user: PropTypes.shape({ username: PropTypes.string }),
+}
+ImpactDialog.defaultProps = {
+  onClose: () => {},
+  buttonOnClick: () => {},
```

Comment:
Given the validation/throw suggestion above, we might want these to default to `undefined`.